rest/route/task_generate.go

```diff
@@ -92,22 +93,28 @@ func (h *generatePollHandler) Parse(ctx context.Context, r *http.Request) error
 }
 
 func (h *generatePollHandler) Run(ctx context.Context) gimlet.Responder {
-	finished, jobErrs, err := data.GeneratePoll(ctx, h.taskID, h.queue)
+	finished, jobErr, err := data.GeneratePoll(ctx, h.taskID, h.queue)
 	if err != nil {
 		grip.Error(message.WrapError(err, message.Fields{
 			"message": "error polling for generated tasks",
 			"task_id": h.taskID,
 		}))
 		return gimlet.MakeJSONInternalErrorResponder(err)
 	}
-	shouldExit := false
-	if len(jobErrs) > 0 { // exit early if we know the error will keep recurring
-		jobErr := errors.New(strings.Join(jobErrs, ", "))
-		shouldExit = db.IsDocumentLimit(jobErr)
+
+	// Exit early if we know the error will keep recurring.
+	// If the parser project is already too big it's not going to get smaller.
+	// If new tasks create a dependency cycle it's going to persist across retries.
+	shouldExit := db.IsDocumentLimit(errors.New(jobErr)) || strings.Contains(jobErr, model.DependencyCycleError.Error())
+
+	var errors []string
+	if len(jobErr) > 0 {
+		errors = append(errors, jobErr)
 	}
+
 	return gimlet.NewJSONResponse(&apimodels.GeneratePollResponse{
 		Finished:   finished,
 		ShouldExit: shouldExit,
-		Errors:     jobErrs,
+		Errors:     errors,
```

Comment:
It seems worth maybe adding a new field to GeneratePollResponse for Error (as opposed to Errors) and sending/processing both for now and then removing Errors, since it's not really necessary to have an errors list, but could just ticket this

Reply:
I started us off with the extra field and opened [EVG-16977](https://jira.mongodb.org/browse/EVG-16977) to remove the deprecated one.